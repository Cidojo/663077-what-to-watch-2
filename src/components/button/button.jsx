import * as React from 'react';
import PropTypes from 'prop-types';
import {SvgIcon} from './../svg-icon/svg-icon.jsx';
import {Icons} from './../../constants.js';

const Button = (props) => {
  const renderIcon = (iconName) => {
    return (
      <SvgIcon {...Icons[iconName]} />
    );
  };

  const {type, className, onClick, icon, children} = props;

  return (
    <button
      type={type}
      className={className}
      onClick={onClick}
    >
      {icon && renderIcon(icon)}
      <span>{children}</span>
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  children: PropTypes.string,
  icon: PropTypes.string,
  onClick: PropTypes.func.isRequired
};

Button.defaultProps = {
  type: `button`,
  className: ``,
  onClick: () => {}
};

export {Button};
