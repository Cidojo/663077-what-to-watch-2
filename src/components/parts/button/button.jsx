import * as React from 'react';
import * as PropTypes from 'prop-types';

import {SvgIcon} from './../svg-icon/svg-icon.jsx';
import {IconProps} from '../../../constants/constants.js';

const Button = (props) => {
  const {type, className, onClick, icon, children} = props;

  return (
    <button
      type={type}
      className={className}
      onClick={onClick}
    >
      {icon && <SvgIcon {...IconProps[icon]} />}
      <span>{children}</span>
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  children: PropTypes.string,
  icon: PropTypes.string,
  onClick: PropTypes.func.isRequired
};

Button.defaultProps = {
  type: `button`,
  className: ``,
  onClick: () => {}
};

export {Button};
