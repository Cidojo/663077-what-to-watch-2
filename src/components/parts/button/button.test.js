import * as React from 'react';
import {Button} from './button.jsx';
import * as renderer from 'react-test-renderer';

const props = {
  type: `button`,
  className: ``,
  onClick: () => {}
};

it(`should render Button component without errors`, () => {
  const tree = renderer.create(<Button {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
