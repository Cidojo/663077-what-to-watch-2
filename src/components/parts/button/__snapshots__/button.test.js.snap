// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button component without errors 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  <span />
</button>
`;
